.github/workflows/test.yml

```diff
@@ -7,8 +7,29 @@ on:
     - "v*"
 
 jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Get commits
+      id: commits
+      run: |
+        echo "MAIN=$(git show -s --format="%H" origin/main)" >> $GITHUB_OUTPUT
+        echo "TAG=$(git rev-list -n 1 ${GITHUB_REF#refs/*/})" >> $GITHUB_OUTPUT
+    - name: Print commits
+      run: |
+        echo "Main commit: ${{ steps.commits.outputs.MAIN }}"
+        echo "Tag commit: ${{ steps.commits.outputs.TAG }}"
+    - name: Compare commits
+      if: ${{ steps.commits.outputs.MAIN != steps.commits.outputs.TAG }}
+      uses: actions/github-script@d556feaca394842dc55e4734bf3bb9f685482fa0
+      with:
+        script: |
+            core.setFailed('Tagged commit does not match main')
```

Comment:
@nikitastupin would you mind taking a look at this? some context: we'd like to release by pushing a tag like `v0.9.0`, but we only want that to happen on the `main` protected branch, and not any other branch. here we're checking the commit to be released against `main`. would you approve?

Reply:
Hi @martriay looks safe to me. Though I'm not sure why two workflows ("Test" and "Build and Release") are needed. Looks like if they were in one it would be easier to read and maintain.

Reply:
The `if` condition already evaluates an expression inside so

```
if: ${{ github.event.workflow_run.conclusion == 'success' }}
```

can be written as

```
if: github.event.workflow_run.conclusion == 'success'
```

Reply:
> Hi @martriay looks safe to me. Though I'm not sure why two workflows ("Test" and "Build and Release") are needed. Looks like if they were in one it would be easier to read and maintain.

i almost explain but thought i didn't want to confuse you. Coverage is the regular PR flow "testing", whereas Test is a sanity check (and condition) at the beginning of the Release workflow. we might need to rename "Testing" to "Pre-release tests" or similar